src/core/utils/urlutils.js

```diff
@@ -59,3 +59,24 @@ export function addParamsToUrl (url, params = {}) {
 export function urlWithoutQueryParamsAndHash (url) {
   return url.split('?')[0].split('#')[0];
 }
+
+/**
+ * returns if two SearchParams objects have the same key,value entries
+ * @param {SearchParams} params1
+ * @param {SearchParams} params2
+ * @return {boolean} true if params1 and params2 have the same key,value entries, false otherwise
+ */
+export function equivalentParams (params1, params2) {
+  if (params1.entries().length !== params2.entries().length) {
+    return false;
+  }
+  for (let entry of params1.entries()) {
```

Comment:
we could probably use Array.some() to be a little more concise here

Reply:
Native `URLSearchParams` returns an iterator (not an Array). This would mean we would need to cast it into an Array first. I'm not sure that work is beneficial for conciseness. What do you think?

Reply:
Oh okay that's a good point. I think you can still make it more concise by removing the 

```js
if (!params2.has(entry[0])) {
  return false;
}
```

if params2 does not have the key entry[0] it will return null, so since entry[1] will always be a string we can skip that check. (nits: maybe we can also change entry to be [key, value], and make it const instead of let?)